### DISCUSSION TITLE
Levelbody Datenpunktwert verändern

### DISCUSSION THREAD

**Discussion by Bernd9000** [Zefau/ioBroker.jarvis#1598] [Q&A]
Kann ich über Datenpunkt Eigenschaften beim Levelbody auch Text ausgeben ?
Mein Dimmer braucht den Helligkeitswert in folgender Form als string im Datenpunkt:
`{"brightness":5}`

**Comment by mcuiobroker** [comment]
ja. Zeig mal den DP, wo es rein muss.

**Comment by Bernd9000** [comment]
`v3.0.11`
Experimentiert hab ich schon mit folgendes um zu testen ob überhaupt Text eingetragen wird aber das klappt nicht:
{"value":"val => 'brightness:  ' +String(val)"}
Es wird nur in jarvis angezeigt aber nicht im ioBroker Datenpunkt eingetragen.
"Component Options" ist auf "verarbeiteter Wert value" gestellt.

![1](https://user-images.githubusercontent.com/77005685/153231647-34081684-9b1a-42b8-9d0a-f2d02febbec6.jpg)

**Reply by mcuiobroker** [reply]
Müsste dann ja ungefähr so aussehen:
`{"value":"val => '{\"brightness: ' +String(val)\"}"}`

**Comment by Bernd9000** [comment]
Klappt nicht, da wird wieder nur der Zahlenwert val eingetragen.
Ich glaube das geht mit dem LevelBody nicht.

**Reply by mcuiobroker** [reply]
Das glaube ich auch. Der schreibt nur den ausgewählten Wert in den DP.
Dann evtl Feature Request?

**Comment by Bernd9000** [comment]
Jo, hab mal eins erstellt [#1601](https://github.com/Zefau/ioBroker.jarvis/issues/1601)
Damit es in jarvis richtig angezeigt wird ist dies übrigens der richtige Wert:
`{"value":"val => '{\"brightness\":' +String(val) +'}'"}`

**Comment by Bernd9000** [comment]
Wobei, mir fällt gerade ein das man nicht zwei Sachen gleichzeitig unter Datenpunkt Eigenschaften angeben kann, oder?
Unter `mqtt.0.shellies.Shelly-Dimmer-Reserve.light.0.status` gibt das Gerät den eingestellten Helligkeitswert zurück.
Den Wert wollte ich mit `{"value":"val => JSON.parse(val).brightness"}` einlesen.

**Comment by Zefau** [comment]
Ist es nicht einfacher über einen extra Datenpunkt in ioBroker zu steuern?

Über jarvis wäre es ein Feature Request aus meiner Sicht. Ich müsste `setOn` / `setOff` so ändern, dass auch Callbacks möglich sind und nicht nur feste Werte.

Zur Erinnerung
- `value` transformiert nur eingehende Wert (von ioBroker nach jarvis)
- `on` / `off` mappt ebenfalls nur eingehende Werte
- `setOn` / `setOff` mappt ausgehende Werte (von jarvis nach ioBroker)

**Comment by Bernd9000** [comment]
@Zefau 
Was heißt in diesen Fällen "mappt" ?
Wenn ich in den Datenpunkt Eigenschaften `{"on":"Ein","off":"Aus"}` rein schreibe wird das ebenfalls als ausgehender Wert (jarvis nach ioBroker) gesendet.Mir scheint es hier ein fließenden Übergang zwischen `"setOn"` und `"on"` zu geben.
Oder ist das zusätzlich vom Gewerk oder von anderen Bedingungen abhängig?

**Comment by Bernd9000** [comment]
Hier mal ein Test mit string als Datenpunkt und verschiedene Werte mit `"on" und "off"`
Im Costum Text steht was ich als Datenpunkt Eigenschaft eingetragen habe.
Im Secondary State was im DP geschrieben wird.
Das heißt wohl es gibt Ausnahmen ?

![Animation](https://user-images.githubusercontent.com/77005685/153451560-ddbb88a5-356d-4f17-ba0f-4a35bb8d4761.gif)
